pyccel/parser/semantic.py

```diff
@@ -2345,12 +2345,14 @@ def _visit_Return(self, expr, **settings):
         return_vars = self.get_function(f_name).results
         assigns     = []
         for v,r in zip(return_vars, results):
-            if not (isinstance(r, PyccelSymbol) and r == v):
+            if not (isinstance(r, PyccelSymbol) and r == (v.name if isinstance(v, Variable) else v)):
```

Comment:
why is the type of v  `Variable`, the nodes are not annotated yet no?

Reply:
`v` is usually a `PyccelSymbol` but if the results are specified (e.g. in the case of a recursive function) then `v` is a `Variable` created [here](https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L2531-L2541)

Reply:
ah okay